src/source/Ice/IceAgent.c

```diff
@@ -1774,6 +1774,61 @@ STATUS iceAgentCheckConnectionStateSetup(PIceAgent pIceAgent)
     return retStatus;
 }
 
+STATUS updateCandidateStats(PIceAgent pIceAgent, BOOL isRemote)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    CHK(pIceAgent != NULL && pIceAgent->pDataSendingIceCandidatePair != NULL, STATUS_NULL_ARG);
+    PIceCandidate pIceCandidate = pIceAgent->pDataSendingIceCandidatePair->remote;
+    PRtcIceCandidateDiagnostics pRtcIceCandidateDiagnostics = &pIceAgent->rtcSelectedRemoteIceCandidateDiagnostics;
+    if (!isRemote) {
+        pIceCandidate = pIceAgent->pDataSendingIceCandidatePair->local;
+        pRtcIceCandidateDiagnostics = &pIceAgent->rtcSelectedLocalIceCandidateDiagnostics;
+        STRNCPY(pRtcIceCandidateDiagnostics->url, STATS_NOT_APPLICABLE_STR, ARRAY_SIZE(pRtcIceCandidateDiagnostics->url));
+        // URL and relay protocol are populated only for local candidate by spec
+        if (pIceCandidate->iceCandidateType == ICE_CANDIDATE_TYPE_HOST) {
+            STRNCPY(pRtcIceCandidateDiagnostics->url, pIceAgent->iceServers[pIceCandidate->iceServerIndex].url,
```

Comment:
what's the meaning of iceServer urls for host candidates?

Reply:
host candidates arent obtained from iceServers. I think NULL or empty string is more appropriate.

Reply:
I changed it to check for pIceCandidate->iceCandidateType != ICE_CANDIDATE_TYPE_HOST. By default, URL is set to N/A.